### PR TITLE
fix(modules): Separate AngularFire services into modules

### PR DESCRIPTION
**Pull request by abehaskins**
Modularizes internal shenanigans

### CODE REVIEW

**Comment by davideast** [comment]
LGTM!

**Comment by abehaskins** [comment]
:goat:

File: src/module.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+  angular.module("firebase.auth", ["firebase.utils"]);
+  angular.module("firebase.database", ["firebase.utils"]);
 
-})(window);
\ No newline at end of file
+  // Define the `firebase` module under which all AngularFire
+  // services will live.
+  angular.module("firebase", ["firebase.utils", "firebase.config", "firebase.auth", "firebase.database"])
+    //TODO: use $window
+    .value("Firebase", exports.firebase)
```
Comment (jwngr) [inline src/module.js]: Why are we still exporting `Firebase` with a capital `"F"`? This should not be required anymore.

File: src/firebase.js [hunk 2]
```diff
@@ -6,6 +6,7 @@
     /** @deprecated */
     .factory("$firebase", function() {
       return function() {
+        //TODO: Update this error to speak about new module stuff
```
Comment (jwngr) [inline src/firebase.js]: Unresolved `TODO`.

File: tests/unit/FirebaseAuth.spec.js [hunk 3]
```diff
@@ -8,8 +8,9 @@ describe('FirebaseAuth',function(){
     log = {
       warn:[]
     };
-
-    module('firebase',function($provide){
+    //
```
Comment (jwngr) [inline tests/unit/FirebaseAuth.spec.js]: These next two lines should be removed.

File: tests/unit/firebaseRef.spec.js [hunk 4]
```diff
@@ -2,9 +2,10 @@
 describe('firebaseRef', function () {
 
   var $firebaseRefProvider;
+  //TODO: Load this from env
```
Comment (jwngr) [inline tests/unit/firebaseRef.spec.js]: Let's just do it :)

File: src/module.js (hunk 1 above)
Comment (abehaskins) [inline src/module.js]: If we removed this it would be a breaking change which we're avoiding here.

File: src/firebase.js (hunk 2 above)
Comment (abehaskins) [inline src/firebase.js]: This TODO is wrong

File: tests/unit/FirebaseAuth.spec.js (hunk 3 above)
Comment (abehaskins) [inline tests/unit/FirebaseAuth.spec.js]: :|

File: tests/unit/firebaseRef.spec.js (hunk 4 above)
Comment (abehaskins) [inline tests/unit/firebaseRef.spec.js]: yeah wasn't quite sure how since spec runs in browser.

File: src/module.js (hunk 1 above)
Comment (jwngr) [inline src/module.js]: Makes sense. Let's at least add a `TODO` comment here to remove this line when the `3.0.0` release happens.

File: tests/unit/firebaseRef.spec.js (hunk 4 above)
Comment (jwngr) [inline tests/unit/firebaseRef.spec.js]: The unit tests run in Node.js. The e2e tests are the only thing that run in a browser environment I believe. So you should just be able to use `process.env` here.

**Pull request merged**